instrumentation/net/http/otelhttp/common.go

```diff
@@ -31,10 +31,17 @@ const (
 
 // Server HTTP metrics.
 const (
-	RequestCount          = "http.server.request_count"           // Incoming request count total
-	RequestContentLength  = "http.server.request_content_length"  // Incoming request bytes total
-	ResponseContentLength = "http.server.response_content_length" // Incoming response bytes total
-	ServerLatency         = "http.server.duration"                // Incoming end to end duration, milliseconds
+	ServerRequestCount          = "http.server.request_count"           // Incoming request count total
+	ServerRequestContentLength  = "http.server.request_content_length"  // Incoming request bytes total
+	ServerResponseContentLength = "http.server.response_content_length" // Incoming response bytes total
+	ServerLatency               = "http.server.duration"                // Incoming end to end duration, milliseconds
```

Comment:
I wonder if these constants need to be public? Technically renaming them is a breaking change. I think the smaller the exposed API surface it, the better it is. WDYT?

Reply:
Oh that is a good point, I forget that other go modules could be relying on these public constants. I don't see why they need to be public though. I'm only familiar with the gRPC metrics and those definitely don't follow this pattern.

Reply:
Another data point here - this module (`go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`) is at `v0.46.1` so making breaking changes is still allowed. So we better get it in shape before it reaches 1.0. This includes hiding/removing everything that doesn't need to be exposed, renaming metrics/method/etc, changing units, etc. So now might be a good time as any to make these private, while it's not too late.

Reply:
At the moment there is global instrumentration vs semconv naming mismath, e.g. "http.client.duration" vs "http.client.request.duration", "http.server.duration" vs "http.server.request.duration" and so on.

Spec: https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#metric-httpclientrequestduration

Will we drop semconv, is it outdated?

Reply:
These constants should live in `semconv` packages. See: https://github.com/open-telemetry/opentelemetry-go/issues/4528

I would rather not export it in `otelhttp` as I see no big value and it would increase the API surface.